src/mame/misc/coolpool.cpp

```diff
@@ -1039,36 +1103,21 @@ ROM_END
  *
  *************************************/
 
-void coolpool_state::machine_start()
-{
-	// assumes it can make an address mask with m_dsp_rom.length() - 1
-	assert(!(m_dsp_rom.length() & (m_dsp_rom.length() - 1)));
-
-	save_item(NAME(m_oldx));
-	save_item(NAME(m_oldy));
-	save_item(NAME(m_result));
-	save_item(NAME(m_lastresult));
-
-	save_item(NAME(m_iop_romaddr));
-}
-
-
-
-void coolpool_state::init_amerdart()
+void amerdart_state::init_amerdart()
 {
 	m_lastresult = 0xffff;
 }
```

Comment:
Does this really need to be a driver initialisation function?  AmeriDarts is the only machine configuration in `amerdart_state`, so you could simply override `machine_start` and set the initial value.